src/graphql/outlier.rs

```diff
@@ -586,3 +609,161 @@ fn latest_outlier_key(before: &str) -> Result<Vec<u8>> {
+
+    let mut connection = Connection::with_additional_fields(
+        has_previous,
+        has_next,
+        RankedOutlierTotalCount {
+            model_id,
+            timestamp,
+        },
+    );
+    connection
+        .edges
+        .extend(nodes.into_iter().map(|(k, ev)| Edge::new(k, ev)));
+    Ok(connection)
+}
+
+#[allow(clippy::type_complexity)] // since this is called within `load` only
+fn load_nodes_with_search_filter<'m, M, I>(
+    map: &'m M,
+    filter: &Option<SearchFilterInput>,
+    after: Option<String>,
+    before: Option<String>,
+    first: Option<usize>,
+    last: Option<usize>,
+) -> Result<(Vec<(String, RankedOutlier)>, bool, bool)>
+where
+    M: IterableMap<'m, I>,
+    I: Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'm,
+{
+    if let Some(last) = last {
+        let iter = if let Some(before) = before {
+            let end = latest_key(&before)?;
+            map.iter_from(&end, Direction::Reverse)?
+        } else {
+            map.iter_backward()?
+        };
+
+        let (nodes, has_more) = if let Some(after) = after {
+            let to = earliest_key(&after)?;
+            iter_through_search_filter_nodes(iter, &to, cmp::Ordering::is_ge, filter, last)
+        } else {
+            iter_through_search_filter_nodes(iter, &[], always_true, filter, last)
+        }?;
+        Ok((nodes, has_more, false))
+    } else {
+        let first = first.unwrap_or(DEFAULT_CONNECTION_SIZE);
+        let iter = if let Some(after) = after {
+            let start = earliest_key(&after)?;
+            map.iter_from(&start, Direction::Forward)?
+        } else {
+            map.iter_forward()?
+        };
+
+        let (nodes, has_more) = if let Some(before) = before {
+            let to = latest_key(&before)?;
+            iter_through_search_filter_nodes(iter, &to, cmp::Ordering::is_le, filter, first)
+        } else {
+            iter_through_search_filter_nodes(iter, &[], always_true, filter, first)
+        }?;
+        Ok((nodes, false, has_more))
+    }
+}
+
+fn iter_through_search_filter_nodes<I>(
+    iter: I,
+    to: &[u8],
+    cond: fn(cmp::Ordering) -> bool,
+    filter: &Option<SearchFilterInput>,
+    len: usize,
+) -> Result<(Vec<(String, RankedOutlier)>, bool)>
+where
+    I: Iterator<Item = (Box<[u8]>, Box<[u8]>)>,
+{
+    let mut nodes = Vec::new();
+    let mut exceeded = false;
+    for (k, v) in iter {
+        if !(cond)(k.as_ref().cmp(to)) {
+            break;
+        }
+
+        let curser = BASE64.encode(&k);
+        let Some(node) = RankedOutlier::from_key_value(&k, &v)?.into() else {
+            continue;
+        };
+
+        if let Some(filter) = filter {
+            if let Some(time) = &filter.time {
+                let start = time.start;
+                let end = time.end;
+                if let Some(start) = start {
+                    if let Some(end) = end {
+                        if node.timestamp < start.timestamp_nanos_opt().unwrap_or_default()
```

Comment:
The comparison condition for the start of the range must be "Greater than or equal"